### PR TITLE
Components: Show empty SectionHeader with a one-line height

### PR DESCRIPTION
**Pull request by jsnajdr**
When rendering an empty `<SectionHeader />`, render it with the same height as if it had a `label` and/or `children`. The empty `SectionHeader` is useful as a placeholder when the content is loading. The `/posts` page displays a similar empty `SectionNav` when the posts are loading.

Implemented by adding a nbsp after the label content (`::after { content: '\00a0' }`). The same technique is used by the `placeholder` mixin.

The first commit in this does a drive-by ES6-ification of the `SectionHeader` component. The real changes are in the second commit.

Before:
section-header-before

After:
section-header-after

Edit: changed the impl to insert the nbsp only when the component is completely empty.
~Does the new extra nbsp after the label have any observable and unwanted effects? The only change I found is that it forces a space between the label and actions elements when they are too close together -- see below.~

Before:
section-header-wrap-before
After:
section-header-wrap-after

### CODE REVIEW

**Comment by folletto** [comment]
Fix wise, it works... but I don't think it's the right approach from an architectural perspective.

What I mean is that I don't think we ever use the component empty by itself. As you mention: "is useful as a placeholder when the content is loading"_ when shown empty is really a loading state for something else.

As such, it should be the controller of this component to trigger a loading state filling, before its content gets loaded.

It could make sense for the component to provide a built-in loading state, in which case I'd do that explicitly in the component, not by injecting a CSS rule that stays there permanently even after the content is loaded.

In short: can we turn this into a state declaration for the component which is in "loading" state until filled, and then replaces the content, thus avoiding to have a leftover CSS in the loaded state?

**Comment by jsnajdr** [comment]
Hello @folletto, thanks for testing this! What I'd like to achieve is a loading UI that looks like the one when posts in `/posts` are loading:

posts-loading

The white rectangle at the top is an empty `<SectionNav />` (rendered [here](https://github.com/Automattic/wp-calypso/blob/6a56111855b66fb795fbdce3b9b874648936a68c/client/my-sites/posts/posts-navigation.jsx#L85-L87)). And there is a post placeholder under it. After the posts load, the `SectionNav` is populated with tab labels and counts:

screen shot 2017-07-28 at 16 29 43

The `SectionNav` supports the empty state and renders it reasonably well. But if I want to use `SectionHeader` for a similar purpose, I need the changes in this PR to make it look good.

It's the controller code of this component that decides whether to render an empty `<SectionHeader />` or whether to pass some content to it.

Does that make sense?

I could make the extra CSS present only if the `SectionHeader` component is in an empty state -- like `SectionNav` does when it [detects that it's empty](https://github.com/Automattic/wp-calypso/blob/master/client/components/section-nav/index.jsx#L80-L93) and adds an `is-empty` CSS class to its element. Is that what you are suggesting?

**Comment by folletto** [comment]
> Is that what you are suggesting?

Correct. 👍 

The specific solution can be CSS, a passed `&nbsp;` as content, or else. The important bit is that it's treated as a separate state that doesn't persist once the load is done :)

**Comment by jsnajdr** [comment]
Changed the implementation to add the CSS `:after` with `&nbsp;` only when the component is really empty -- there's an extra `is-empty` CSS class to mark the empty state. Thanks @folletto for the suggestion!

**Comment by folletto** [comment]
No more to add for design review then 👍
